test-data/unit/fine-grained.test

```diff
@@ -629,6 +629,304 @@ class M(type):
 ==
 a.py:4: error: Incompatible types in assignment (expression has type "str", variable has type "int")
 
+[case testDataclassUpdate1]
```

Comment:
It would be good to have at least one test case with a deeper inheritance hierarchy. For example, have `A` <- `B` <- `C` and test that a change in `A` will get propagated to `C`.